src/rust_tree/traversal.rs

```diff
@@ -83,10 +83,68 @@ fn format_date(time: SystemTime) -> String {
     }
 }
 
+// Helper function to check if a directory (recursively) contains any files matching the pattern
+fn directory_contains_pattern_matches(
+    dir_path: &Path,
+    options: &TreeOptions,
+    depth: usize,
+) -> std::io::Result<bool> {
+    // If no pattern is specified, assume there are matches
+    let pattern = match &options.pattern_glob {
+        Some(p) => p,
+        None => return Ok(true),
+    };
+
+    // Check level limit to avoid infinite recursion
+    if let Some(max_level) = options.level {
+        if depth >= max_level as usize {
+            return Ok(false);
+        }
+    }
+
+    let read_dir_result = fs::read_dir(dir_path);
+    let entries = match read_dir_result {
+        Ok(reader) => reader.filter_map(Result::ok).collect::<Vec<_>>(),
+        Err(_) => return Ok(true), // If we can't read the directory, include it to show error message
```

Comment:
smh. this completely breaks pattern filtering. Failures now return `true` (matches pattern). instead of false.
If we can't read the dir, why would we assume it matches??